File: myproj27/main03.js
const { melon_data: song_array } = require("./melon_data");


// TODO: #3 좋아요수 top10을 출력
// Array의 sort 활용
// 출력포맷 : `[좋아요수] 곡명 가수명`


const top10_song_array = song_array.sort(
    (song1, song2) => song2.like - song1.like
).slice(0, 10);


for (const [index, song] of top10_song_array.entries()) {
    const rank = index + 1;
    console.log(`${rank} [${song.like}] ${song.title} by ${song.artist}`);
}

File: myproj27/main03.py
import json

# 한글 윈도우는 기본 인코딩이 cp949이기에
# open을 통한 text 파일 열기에서는 필히 encoding을 지정해주어야 합니다.
with open("melon_data.json", "rt", encoding="utf-8") as f:
    song_list = json.loads(f.read())

song_list.sort(key=lambda song: song['like'], reverse=True)

for (index, song) in enumerate(song_list[:10], 1):
    print("{index} [{like}] {title} by {artist}".format(index=index, **song))
